### DISCUSSION TITLE
第2节课作业(2班)

### DISCUSSION THREAD

**Discussion by MING-ZCH** [InternLM/Tutorial#103] [作业]
# 在下方讨论区提交作业（图片/链接形式均可）~
## 基础作业：
* 使用 InternLM-Chat-7B 模型生成 300 字的小故事（需截图）。
* 熟悉 hugging face 下载功能，使用 huggingface_hub python 包，下载 InternLM-20B 的 config.json 文件到本地（需截图下载过程）。

## 进阶作业（可选做）
* 完成浦语·灵笔的图文理解及创作部署（需截图）
* 完成 Lagent 工具调用 Demo 创作部署（需截图）

**Comment by tcexeexe** [comment]
### 基础作业：
## 1：使用 InternLM-Chat-7B 模型生成 300 字的小故事
```
很好，这是一个关于勇气和冒险的故事：

有一天，一个小男孩叫做汤姆，决定去探险。他穿上了他的衣服，带上他的背包，离开了家，开始了他的冒险之旅。他穿过森林，越过小溪，爬过山脉，最终到达了一个神秘的洞穴。

在洞穴里，汤姆发现了一个神秘的宝藏。他开始挖掘，但很快就遇到了困难。洞穴的墙壁很坚固，他无法轻易地打破它们。但是，他并没有放弃。他继续挖掘，直到他找到了一个秘密通道。

在通道里，汤姆发现了一个神秘的宝藏。他开始挖掘，但很快就遇到了困难。洞穴的墙壁很坚固，他无法轻易地打破它们。但是，他并没有放弃。他继续挖掘，直到他找到了一个秘密通道。

在通道里，汤姆发现了一个神秘的宝藏。他开始挖掘，但很快就遇到了困难。洞穴的墙壁很坚固，他无法轻易地打破它们。但是，他并没有放弃。他继续挖掘，直到他找到了一个秘密通道。

在通道里，汤姆发现了一个神秘的宝藏。他开始挖掘，但很快就遇到了困难。洞穴的墙壁很坚固，他无法轻易地打破它们。但是，他并没有放弃。他继续挖掘，直到他找到了一个秘密通道。

在通道里，汤姆发现了一个神秘的宝藏。他开始挖掘，但很快就遇到了困难。洞穴的墙壁很坚固，他无法轻易地打破它们。但是，他并没有放弃。他继续挖掘，直到他找到了一个秘密通道。

在通道里，汤姆发现了一个神秘的宝藏。他开始挖掘，但很快就遇到了困难。洞穴的墙壁很坚固，他无法轻易地打破它们。但是，他并没有放弃。他继续挖掘，直到他找到了一个秘密通道。

在通道里，汤姆发现了一个神秘的宝藏。他开始挖掘，但很快就遇到了困难。洞穴的墙壁很坚固，他无法轻易地打破它们。但是，他并没有放弃。他继续挖掘，直到他找到了一个秘密通道。

在通道里，汤姆发现了一个神秘的宝藏。他开始挖掘，但很快就遇到了困难。洞穴的墙壁很坚固，他无法轻易地打破它们。但是，他并没有放弃。他继续挖掘，直到他找到了一个秘密通道。

在通道里，汤姆发现了一个神秘的宝藏。他开始挖掘，但很快就遇到了困难。洞穴的墙壁很坚固，他无法轻易地打破它们。但是，他并没有放弃。他继续挖掘，直到他找到了一个秘密通道。

在通道里，汤姆发现了一个神秘的宝藏。他开始挖掘，但很快就遇到了困难。洞穴的墙壁很坚固，他无法轻易地打破它们。但是，他并没有放弃。他继续挖掘，直到他找到了一个秘密通道。

在通道里，汤姆发现了一个神秘的宝藏。他开始挖掘，但很快就遇到了困难。洞穴的墙壁很坚固，他无法轻易地打破它们。但是，他并没有放弃。他继续挖掘，直到他找到了一个秘密通道。

在通道里，汤姆发现了一个神秘的宝藏。他开始挖掘，但很快就遇到了困难。洞穴的墙壁很坚固，他无法轻易地打破它们。但是，他并没有放弃。他继续挖掘，直到他找到了一个秘密通道。

在通道里，汤姆发现了一个神秘的宝藏。他开始挖掘，但很快就遇到了困难。洞穴的墙壁很坚固，他无法轻易地打破它们。但是，他并没有放弃。他继续挖掘，直到他找到了一个秘密通道。

在通道里，汤姆发现了一个神秘的宝藏。他开始挖掘，但很快就遇到了困难。洞穴的墙壁很坚固，他无法轻易地打破它们。但是，他并没有放弃。他继续挖掘，直到他找到了一个秘密通道。

在通道里，汤姆发现了一个神秘的宝藏。他开始挖掘，但很快就遇到了困难。洞穴的墙壁很坚固，他无法轻易地打破它们。但是，他并没有放弃。他继续挖掘，直到他找到了一个秘密通道。

在通道里，汤姆发现了一个神秘的宝藏。他开始挖掘，但很快就遇到了困难。洞穴的墙壁很坚固，他无法轻易地打破它们。但是，他并没有放弃。他继续挖掘，直到他找到了一个秘密通道。

在通道里，汤姆发现了一个神秘的宝藏。他开始挖掘，但很快就遇到了困难。洞穴的墙壁很坚固，他无法轻易地打破它们。但是，他并没有放弃。他继续挖掘，直到他找到了一个秘密通道。

在通道里，汤姆发现了一个神秘的宝藏。他开始挖掘，但很快就遇到了困难。洞穴的墙壁很坚固，他无法轻易地打破它们。但是，他并没有放弃。他继续挖掘，直到他找到了一个秘密通道。

在通道里，汤姆发现了一个神秘的宝藏。他开始挖掘，但很快就遇到了困难。洞穴的墙壁很坚固，他无法轻易地打破它们。但是，他并没有放弃。他继续挖掘，直到他找到了一个秘密通道。

在通道里，汤姆发现了一个神秘的宝藏。他开始挖掘，但很快就
```
<img width="828" alt="1704511317015" src="https://github.com/InternLM/tutorial/assets/28287196/0f61ec26-c69d-49f8-832b-8b2408fb88c0">

## 2：下载 InternLM-20B 的 config.json 文件到本地（需截图下载过程）
<img width="806" alt="1704511584632" src="https://github.com/InternLM/tutorial/assets/28287196/dc22dff9-a258-41e0-849f-02140d40ee5e">

**Comment by YanxingLiu** [comment]
1. 使用 InternLM-Chat-7B 模型生成 300 字的小故事
<img width="841" alt="image" src="https://github.com/InternLM/tutorial/assets/42299757/40b326cc-8c6c-49f6-b8c6-aa7c59efa6c6">

2. 熟悉 hugging face 下载功能，使用 huggingface_hub python 包，下载 InternLM-20B 的 config.json 文件到本地
<img width="934" alt="image" src="https://github.com/InternLM/tutorial/assets/42299757/82adcff9-83d7-439d-b7b3-c9e28084a530">

3. 完成浦语·灵笔的图文理解及创作部署（需截图）
<img width="1600" alt="image" src="https://github.com/InternLM/tutorial/assets/42299757/647ebffc-70c1-4aad-880e-cf3bdf7e7d77">

4. 完成 Lagent 工具调用 Demo 创作部署（需截图）
<img width="1467" alt="image" src="https://github.com/InternLM/tutorial/assets/42299757/382e179d-a60f-4782-ab3c-1245daceaa46">

**Comment by scuizhibin** [comment]
![sss](https://github.com/InternLM/tutorial/assets/32003970/dd76bd75-ed8b-4590-b81c-2f26604de54b)

**Reply by MING-ZCH** [reply]
同学，请在群公告信息收集表中修改你的GitHub名（与当前名称scuizhibin一致），方便后续打分~
并填写链接

**Comment by kishiokon** [comment]
使用 InternLM-Chat-7B 模型生成 300 字的小故事（需截图）。
![Y`2O15@FC0I_A5( 9XZW44](https://github.com/InternLM/tutorial/assets/92700495/ee8c53c1-1a46-4d10-a9dc-7050fee7df6b)

熟悉 hugging face 下载功能，使用 huggingface_hub python 包，下载 InternLM-20B 的 config.json 文件到本地（需截图下载过程）。
![L8591YX}S4%EOXF2~QL~4DY](https://github.com/InternLM/tutorial/assets/92700495/edd63659-3cbe-4155-9e2b-53903f8deacd)

完成浦语·灵笔的图文理解及创作部署（需截图）
![%R646JC5Y OYSWLG8S 03CX](https://github.com/InternLM/tutorial/assets/92700495/01742c47-69d2-4779-b66e-edcf69381893)
![~H1D$1_0LDW}}W@0BLQB E5](https://github.com/InternLM/tutorial/assets/92700495/ec75434f-7691-4608-a561-75ac6de5f8ce)

完成 Lagent 工具调用 Demo 创作部署（需截图）
![%04) `4M_4%F47Q)GE0L{%1](https://github.com/InternLM/tutorial/assets/92700495/a7852175-3408-4afa-978f-ae44f39abdf9)

**Comment by songyuc** [comment]
1、使用 InternLM-Chat-7B 模型生成 300 字的小故事
<img width="1274" alt="8eadc17f40f200b2a6d23185014e9fa" src="https://github.com/InternLM/tutorial/assets/27288110/f82fc6da-bca8-478d-a562-2cf290c41a4f">
2、熟悉 hugging face 下载功能，使用 huggingface_hub python 包，下载 InternLM-20B 的 config.json 文件到本地
![image](https://github.com/InternLM/tutorial/assets/27288110/02e55c1a-2d28-41f0-9175-c4933a5ed88a)
4、完成 Lagent 工具调用 Demo 创作部署

**Comment by linmiaoyan** [comment]
笔记、插件体验、灵笔体验、结合语料测试 （是bug吗？）
![p1](https://github.com/InternLM/tutorial/assets/50226939/5f6b74c4-78b7-4aa8-b1e3-51d716915b83)
![2](https://github.com/InternLM/tutorial/assets/50226939/91ce0863-dc5a-452f-b132-0fabedc6bd7e)
![3](https://github.com/InternLM/tutorial/assets/50226939/44795494-7836-4f8f-b95a-c2b5a039039e)
![4](https://github.com/InternLM/tutorial/assets/50226939/85f79426-59d5-446c-9f09-2d8d4255eff1)
![5](https://github.com/InternLM/tutorial/assets/50226939/885d8632-9abe-46d7-86a5-6a7b0ba0afcd)

**Comment by pranerd** [comment]
### 基础作业：
使用 InternLM-Chat-7B 模型生成 300 字的小故事
![interlm_300](https://github.com/InternLM/tutorial/assets/19358928/531133fe-9c55-44b8-8b47-1551ad7befc5)

熟悉 hugging face 下载功能，使用 huggingface_hub python 包，下载 InternLM-20B 的 config.json 文件到本地（需截图下载过程）
![interlm_hugging_download](https://github.com/InternLM/tutorial/assets/19358928/cfea628e-ab1b-4596-8d0f-711fccabcac0)

### 进阶作业（可选做）
完成浦语·灵笔的图文理解及创作部署
![interlm_xcomposer_1](https://github.com/InternLM/tutorial/assets/19358928/881bf9ad-8cc9-446a-bc6f-6ebf9a0bc171)
![interlm_xcomposer_2](https://github.com/InternLM/tutorial/assets/19358928/9ac9770b-931f-4ebc-8227-d2ccb65dd3ca)

完成 Lagent 工具调用 Demo 创作部署
![interlm_lagent_1](https://github.com/InternLM/tutorial/assets/19358928/d27b3775-f54d-48fc-bc00-8d98e48e8182)
![interlm_lagent_2](https://github.com/InternLM/tutorial/assets/19358928/1ead2d2d-4a9c-4f36-8d13-1e6da14aa1de)
![interlm_lagent_3](https://github.com/InternLM/tutorial/assets/19358928/3bde125c-ba54-40e0-a424-bddb90d29b33)

**Comment by kyrie2to11** [comment]
[InterLM L-2: 轻松玩转书生·浦语大模型趣味 Demo](https://zhuanlan.zhihu.com/p/676498904?) 打卡~

**Comment by xrxu** [comment]
https://github.com/xrxu/Docu/blob/master/helloworld/helloworld.md

**Reply by xrxu** [reply]
第二课作业

**Comment by DarthThomas** [comment]
# 基础作业
- 使用 InternLM-Chat-7B 模型生成 300 字的小故事
<img width="776" alt="截屏2024-01-07 17 15 00" src="https://github.com/InternLM/tutorial/assets/20266717/f509c3ed-6f5a-447e-b733-d6e6c45b2895">
    
- 熟悉 hugging face 下载功能，使用 huggingface_hub python 包，下载 InternLM-20B 的 config.json 文件到本地
<img width="776" alt="截屏2024-01-07 18 57 33" src="https://github.com/InternLM/tutorial/assets/20266717/ffc67949-75e4-481e-a3bd-e1a39e5ddf8b">

**Comment by qihong2019** [comment]
使用 InternLM-Chat-7B 模型生成 300 字的小故事
![image](https://github.com/InternLM/tutorial/assets/47970679/ed72e5c8-9bd4-4138-a251-5dd666009e6d)

熟悉 hugging face 下载功能
![image](https://github.com/InternLM/tutorial/assets/47970679/76e3232c-ed91-489b-bfd9-13cca862aac5)

完成浦语·灵笔的图文理解及创作部署
![image](https://github.com/InternLM/tutorial/assets/47970679/e2de06df-0e0b-4e3d-8fbd-afd5cad9d564)

完成 Lagent 工具调用 Demo 创作部署
![image](https://github.com/InternLM/tutorial/assets/47970679/c8309b43-0868-4e67-a551-2f84ef9da5d6)

**Comment by enkilee** [comment]
[书生·浦语大模型全链路开源体系-Lesson2-mission](https://blog.csdn.net/luhputu520/article/details/135453542)

**Comment by jinxianwei** [comment]
[第二课作业](https://blog.csdn.net/jinxianwei1999/article/details/135443647?csdn_share_tail=%7B%22type%22%3A%22blog%22%2C%22rType%22%3A%22article%22%2C%22rId%22%3A%22135443647%22%2C%22source%22%3A%22jinxianwei1999%22%7D)

**Comment by yangff0016** [comment]
[第二节课作业](https://blog.csdn.net/weixin_42837310/article/details/135466151?spm=1001.2014.3001.5502)

**Comment by CoderJackZhu** [comment]
[第二节作业](https://zhuanlan.zhihu.com/p/676665482)

**Comment by ZixinxinWang** [comment]
基础作业：
1. 
![4eed69e144df4226bdd2ce3b1d72dfcc](https://github.com/InternLM/tutorial/assets/152623177/b7e6520e-20c8-4ffc-ba34-18936669034b)
2.
<img width="733" alt="image" src="https://github.com/InternLM/tutorial/assets/152623177/7e1a979a-58b0-444e-b6f9-8fff91a72878">
进阶作业：
1.
![image](https://github.com/InternLM/tutorial/assets/152623177/890211c4-4c0b-4ccd-a4af-f52be4a5e647)
2.
![image](https://github.com/InternLM/tutorial/assets/152623177/380e9651-3097-4d91-abbe-6f1b0b8fcab9)

**Comment by RyanWu31** [comment]
https://github.com/RyanWu31/tutorial_notebook

**Comment by heiqiyihu-sp** [comment]
基础作业
![微信图片_20240109185615](https://github.com/InternLM/tutorial/assets/73636327/9908c999-5df1-4e67-9488-d95d3ee7a1c2)

**Comment by yanxiaoqi932** [comment]
![20240109154830](https://github.com/InternLM/tutorial/assets/55906038/964d6390-c315-428e-b5af-dd4c3816d055)
![20240109185556](https://github.com/InternLM/tutorial/assets/55906038/516d0c40-5e2d-4f27-829f-0ad4785d6e56)
![20240109160320](https://github.com/InternLM/tutorial/assets/55906038/60a548a4-c7c2-4fb8-bc20-ce48ac109dab)
![20240109162034](https://github.com/InternLM/tutorial/assets/55906038/b6795a7f-d357-4626-9bbc-a9e4676d8f8f)

**Comment by ljqiff** [comment]
# 基础
## 使用 InternLM-Chat-7B 模型生成 300 字的小故事（需截图）。
![image](https://github.com/InternLM/tutorial/assets/29425919/7a99cf47-010d-4f77-8a12-ca3f65e866ca)
![image](https://github.com/InternLM/tutorial/assets/29425919/649ff172-2cf9-4787-8ad4-b8a22766e8a6)

## 熟悉 hugging face 下载功能，使用 huggingface_hub python 包，下载 InternLM-20B 的 config.json 文件到本地（需截图下载过程）。
![image](https://github.com/InternLM/tutorial/assets/29425919/1a05a851-e8f8-4eba-b615-427ec682283b)
# 进阶作业（可选做）
## 完成浦语·灵笔的图文理解及创作部署（需截图）
![image](https://github.com/InternLM/tutorial/assets/29425919/67ffb0bf-2b8f-49ab-90c1-390d02330464)

## 完成 Lagent 工具调用 Demo 创作部署（需截图
![image](https://github.com/InternLM/tutorial/assets/29425919/2db3c5f9-f26b-4fe7-bfd8-30fbf4baba38)

**Comment by dcuplover** [comment]
[第二课轻松玩转书生·浦语大模型趣味](https://www.yuque.com/u174176/new86d/bklqmih3kmwwqu25?singleDoc#)

**Comment by Mr-Vincenyang** [comment]
基础作业：

使用 InternLM-Chat-7B 模型生成 300 字的小故事（需截图）。
![第2节1](https://github.com/InternLM/tutorial/assets/120323418/37db749a-19f1-4c67-9341-0f5d25c93468)

熟悉 hugging face 下载功能，使用 huggingface_hub python 包，下载 InternLM-20B 的 config.json 文件到本地（需截图下载过程）。
<img width="849" alt="第2节2" src="https://github.com/InternLM/tutorial/assets/120323418/7b839586-de69-47e0-b3c6-1ac9bf5cb90f">

进阶作业（可选做）

完成浦语·灵笔的图文理解及创作部署（需截图）
![第2节3](https://github.com/InternLM/tutorial/assets/120323418/135fdda1-f432-40ed-aecf-f1fcf883bccd)

完成 Lagent 工具调用 Demo 创作部署（需截图）
![第2节4](https://github.com/InternLM/tutorial/assets/120323418/861dc9e1-a2e9-4732-8373-e8ff165796f6)

**Comment by WanpengXu** [comment]
因期末考试 ddl 耽误了，请 TA 评阅，谢谢～
[[作业] 书生·浦语大模型实战营-第二节](https://b07ofnm9xj7.feishu.cn/wiki/G259wLRCSi933tkIPlycWfC1nFb)

**Comment by kaoyansoft123** [comment]
![image](https://github.com/InternLM/tutorial/assets/152302734/232a0fb6-f6a1-4316-a4e8-71d83f4460e6)
<img width="851" alt="819c02ee846fa17f3b106dfeb506bff" src="https://github.com/InternLM/tutorial/assets/152302734/b02dd0ea-112c-47e9-9876-04d764bde594">
<img width="1134" alt="259a83b48b213fde02f5d7f9e3a34ee" src="https://github.com/InternLM/tutorial/assets/152302734/09c50314-1247-4737-8b22-26c97310a4fd">
![Uploading 3378b6bcd87676498a52f9e392f2030.png…]()

**Reply by kaoyansoft123** [reply]
<img width="1140" alt="3378b6bcd87676498a52f9e392f2030" src="https://github.com/InternLM/tutorial/assets/152302734/de4481ab-9914-4cf2-bbb0-bf8ce91ffcf1">

**Comment by MING-ZCH** [comment]
基础作业：
1.
![image](https://github.com/InternLM/tutorial/assets/119648793/a540d5c2-0bb0-4f05-aede-84329608aab0)

2.
![image](https://github.com/InternLM/tutorial/assets/119648793/e06d64d9-adf0-4d78-a18d-f5f95885152a)

**Comment by KylinCarol** [comment]
基础作业：
1​​
![a5ad3089514b9f84a742e106a95ffa6](https://github.com/InternLM/tutorial/assets/135086891/60ad1354-9caf-4dba-843c-95f5515b4eee)
2
![fd2431d01012dcc129a39ad55f33fe3](https://github.com/InternLM/tutorial/assets/135086891/2b29d4f3-e154-41e8-ad37-d211d5442a0f)
进阶作业：
1
![image](https://github.com/InternLM/tutorial/assets/135086891/ee025bdd-2dff-46cb-9b59-fba5bc06d098)
2
![image](https://github.com/InternLM/tutorial/assets/135086891/2c1ed522-8f50-45f2-b684-056091accdf4)
![image](https://github.com/InternLM/tutorial/assets/135086891/4920d346-8a9f-48e5-8ead-d079857c6ce7)

**Comment by zyy0831** [comment]
第二课作业 
https://blog.csdn.net/sinat_41838539/article/details/135466375?spm=1001.2014.3001.5501

**Comment by PrinceRay7** [comment]
第二课作业
https://github.com/PrinceRay7/InternLM_HW/blob/main/hw2/homework/homework.md
